mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/restapi/v2/MCRRestObjectAccessKeys.java

```diff
@@ -152,27 +131,23 @@ public Response getAccessKey(@PathParam(PARAM_MCRID) final MCRObjectID objectId,
     @Produces(MediaType.APPLICATION_JSON)
     @MCRRequireAccessKeyAuthorization
     @MCRRequireTransaction
-    public Response createAccessKey(@PathParam(PARAM_MCRID) final MCRObjectID objectId, final String accessKeyJson) {
-        final MCRAccessKey accessKey = MCRAccessKeyTransformer.accessKeyFromJson(accessKeyJson);
-        if (!MCRMetadataManager.exists(objectId)) {
-            throw getUnknownObjectException();
-        }
-        MCRAccessKeyManager.createAccessKey(objectId, accessKey);
-        return Response.created(uriInfo.getAbsolutePathBuilder().path(accessKey.getSecret()).build()).build();
+    public Response createAccessKeyForObject(@PathParam(PARAM_MCRID) final MCRObjectID objectId,
+        final String accessKeyJson) {
+        return MCRRestAccessKeyHelper.doCreateAccessKey(objectId, accessKeyJson, uriInfo);
     }
-    
+
     @PUT
-    @Path("/{" + SECRET + "}")
+    @Path("/{" + PARAM_SECRET + "}")
     @Operation(
-        summary = "Update MCRAccessKey",
+        summary = "Updates an access key for an object",
         responses = {
-            @ApiResponse(responseCode = "204", description = "MCRAccessKey successfully updated"),
-            @ApiResponse(responseCode = "400", description = "Invalid request. Check type, values.",
+            @ApiResponse(responseCode = "204", description = "Access key was successfully updated"),
+            @ApiResponse(responseCode = "400", description = "Invalid ID or invalid access key",
                 content = { @Content(mediaType = MediaType.APPLICATION_JSON) }),
             @ApiResponse(responseCode = "401",
                 description = "You do not have create permission and need to authenticate first",
                 content = { @Content(mediaType = MediaType.APPLICATION_JSON) }),
-            @ApiResponse(responseCode = "404", description = "ObjectID or MCRAccessKey doesn't exists",
+            @ApiResponse(responseCode = "404", description = "Object or access key doesn't exists",
```

Comment:
"Object or access key does not exist"
I am sure - no "s"-ending for exist / I would avoid short forms like "doesn't"

Please check all other 404-messages as well!